connectors/src/connectors/google_drive/index.ts

```diff
@@ -543,51 +614,122 @@ export async function setGoogleDriveConnectorPermissions(
+              },
+              {
+                driveSheetId: s.googleSheetId,
+              },
+            ],
+          })),
+        },
+      })
+    : [];
+
+  const folderOrFileNodes = await concurrentExecutor(
     folderOrFiles,
     async (f): Promise<ContentNode> => {
-      const type = getPermissionViewType(f);
+      const type = getPermissionViewType(f, viewType);
       let sourceUrl = null;
-      if (type === "file") {
-        sourceUrl = `https://drive.google.com/file/d/${f.driveFileId}/view`;
-      } else if (type === "folder") {
-        sourceUrl = `https://drive.google.com/drive/folders/${f.driveFileId}`;
-      } else if (type === "database") {
+
+      if (f.mimeType === "application/vnd.google-apps.spreadsheet") {
         sourceUrl = `https://docs.google.com/spreadsheets/d/${f.driveFileId}/edit`;
+      } else if (f.mimeType === "application/vnd.google-apps.folder") {
+        sourceUrl = `https://drive.google.com/drive/folders/${f.driveFileId}`;
+      } else {
+        sourceUrl = `https://drive.google.com/file/d/${f.driveFileId}/view`;
       }
+
       return {
         provider: "google_drive",
         internalId: f.driveFileId,
         parentInternalId: null,
-        type: getPermissionViewType(f),
+        type,
         title: f.name || "",
         dustDocumentId: getGoogleDriveEntityDocumentId(f),
         lastUpdatedAt: f.lastUpsertedTs?.getTime() || null,
         sourceUrl,
-        expandable:
-          (await GoogleDriveFiles.findOne({
-            attributes: ["id"],
-            where: {
-              connectorId: connectorId,
-              parentId: f.driveFileId,
-            },
-          })) !== null,
+        expandable: await isDriveObjectExpandable({
+          objectId: f.driveFileId,
+          mimeType: f.mimeType,
+          connectorId,
+          viewType,
+        }),
         permission: "read",
+        dustTableId: null,
       };
     },
     { concurrency: 4 }
   );
 
-  return new Ok(nodes);
+  const sheetNodes = await concurrentExecutor(
+    sheets,
+    async (s): Promise<ContentNode> => {
+      return {
+        provider: "google_drive",
+        internalId: getGoogleSheetContentNodeInternalId(
+          s.driveFileId,
+          s.driveSheetId
+        ),
+        parentInternalId: s.driveFileId,
+        type: "database",
+        title: s.name || "",
+        dustDocumentId: null,
+        lastUpdatedAt: s.updatedAt.getTime() || null,
+        sourceUrl: `https://docs.google.com/spreadsheets/d/${s.driveFileId}/edit`,
+        expandable: false,
+        permission: "read",
+        dustTableId: getGoogleSheetTableId(s.driveFileId, s.driveSheetId),
+      };
+    },
+    { concurrency: 4 }
+  );
+
+  // Return the nodes in the same order as the input internalIds.
+  const nodeByInternalId = new Map<string, ContentNode>();
+  for (const node of folderOrFileNodes.concat(sheetNodes)) {
+    nodeByInternalId.set(node.internalId, node);
+  }
```

Comment:
`nit`
```suggestion
  
  const nodeByInternalId = new Map<string, ContentNode>(
 [...folderOrFileNodes, ...sheetNodes].map(n => [n.internalId, n] as [string, ContentNode])
);
```

Reply:
TIL ! In fact, the `as` isn't even need AFAICT

Reply:
And the generics params aren't needed either. This is pretty useful I'm going to use this a lot more

Reply:
🔥 🔥 🔥